contracts/mainnet/connectors/pooltogether/main.sol

```diff
@@ -0,0 +1,194 @@
+pragma solidity ^0.7.0;
+
+/**
+ * @title PoolTogether
+ * @dev Deposit & Withdraw from PoolTogether
+ */
+
+ import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+ import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+ import { PrizePoolInterface, TokenFaucetInterface, TokenFaucetProxyFactoryInterface, PodInterface } from "./interface.sol";
+
+import { TokenInterface } from "../../common/interfaces.sol";
+import { Events } from "./events.sol";
+import { DSMath } from "../../common/math.sol";
+import { Basic } from "../../common/basic.sol";
+
+abstract contract PoolTogetherResolver is Events, DSMath, Basic {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev Deposit into Prize Pool
+     * @notice Deposit assets into the Prize Pool in exchange for tokens
+     * @param prizePool PrizePool address to deposit to
+     * @param amount The amount of the underlying asset the user wishes to deposit. The Prize Pool contract should have been pre-approved by the caller to transfer the underlying ERC20 tokens.
+     * @param controlledToken The address of the token that they wish to mint. For our default Prize Strategy this will either be the Ticket address or the Sponsorship address.  Those addresses can be looked up on the Prize Strategy.
+     * @param getId Get token amount at this ID from `InstaMemory` Contract.
+     * @param setId Set token amount at this ID in `InstaMemory` Contract.
+    */
+
+    function depositTo(
+        address prizePool,
+        uint256 amount,
+        address controlledToken,
+        uint256 getId,
+        uint256 setId
+    ) external payable returns ( string memory _eventName, bytes memory _eventParam) {
+        uint _amount = getUint(getId, amount);
+
+        PrizePoolInterface prizePoolContract = PrizePoolInterface(prizePool);
+        address prizePoolToken = prizePoolContract.token();
+
+        // Approve prizePool
+        TokenInterface tokenContract = TokenInterface(prizePoolToken);
+        _amount = _amount == uint256(-1) ? tokenContract.balanceOf(address(this)) : _amount;
+        tokenContract.approve(prizePool, _amount);
```

Comment:
@eccheung4 Just wanted to confirm - Is there any prize pool which has eth or weth has prizePoolToken? Or there be in future?

Reply:
Currently I don't see any WETH as prizePoolToken, but it is possible to use WETH, as any ERC20 can be used to create a Prize Pool. ETH is not possible.

Reply:
@eccheung4 Then I think, we can add ETH(WETH) cases in all deposit and withdrawal functions. If the staking token is WETH, then we will convert ETH into WETH for deposit functions and convert WETH into ETH for withdrawal functions. So, the connector is easily compatible for WETH case too.

Reply:
@eccheung4 Assuming it's a Prize Pool that accepts WETH, then if an user sends ETH with transaction and has WETH as well, how would I know which to use or I should just assume it'd be the ETH to use?

Reply:
@eccheung4 We can consider, user will have only ETH in DSA wallet. AaveV2 Deposit eg: https://github.com/Instadapp/dsa-connectors/blob/main/contracts/mainnet/connectors/aave/v2/main.sol#L33-L45

Reply:
@thrilok209 I've added logic for converting ETH to WETH for a WETH PrizePool. 
I found a community Prize Pool that used WETH:
https://community.pooltogether.com/pools/mainnet/0xa88ca010b32a54d446fc38091ddbca55750cbfc3/manage#stats